File: library/src/main/java/com/mayz/retrofitutils/schedulers/BaseSchedulerProvider.java
package com.mayz.retrofitutils.schedulers;

import androidx.annotation.NonNull;
import io.reactivex.ObservableTransformer;
import io.reactivex.Scheduler;


public interface BaseSchedulerProvider {

    @NonNull
    Scheduler computation();

    @NonNull
    Scheduler io();

    @NonNull
    Scheduler ui();

    @NonNull
    <T> ObservableTransformer<T, T> applySchedulers();
}

File: kotlinlibrary/src/main/java/com/mayz/kotlinlibrary/schedulers/BaseSchedulerProvider.kt
package com.mayz.kotlinlibrary.schedulers

import androidx.annotation.NonNull
import io.reactivex.ObservableTransformer
import io.reactivex.Scheduler

/**
 * @createDate: 2020/5/8
 * @author:     mayz
 * @version:    1.0
 */
interface BaseSchedulerProvider {
    @NonNull
    fun computation():Scheduler
    @NonNull
    fun io():Scheduler
    @NonNull
    fun ui():Scheduler
    @NonNull
    fun <T> applySchedulers(): ObservableTransformer<T?, T?>?
}